File: WorldFeed/Client/src/app/core/Core/Web/IWebRequestExecutor.ts
import {ITwitterRequest} from "../../Public/Models/Interfaces/ITwitterRequest";
import {ITwitterClientHandler} from "./ITwitterClientHandler";
import {ITwitterResponse} from "./ITwitterResponse";

// Generate a Token that can be used to perform OAuth queries
export interface IWebRequestExecutor {
  // Execute a TwitterQuery and return the resulting json data.
  executeQueryAsync(request: ITwitterRequest, handler?: ITwitterClientHandler): Promise<ITwitterResponse>;

  // Execute a multipart TwitterQuery and return the resulting json data.
  executeMultipartQueryAsync(request: ITwitterRequest): Promise<ITwitterResponse>;
}

File: Sharebook/Server/src/Services/Upload/Sharebook.Upload.API/Web/IWebRequestExecutor.cs
﻿namespace Sharebook.Upload.API.Web
{
    using System.Threading.Tasks;

    using Sharebook.Common.Web;
    using Sharebook.Upload.API.Web;
    using Sharebook.Upload.Application.Requesters;


    /// <summary>
    /// Generate a Token that can be used to perform OAuth queries
    /// </summary>
    public interface IWebRequestExecutor
    {
        /// <summary>
        /// Execute a TwitterQuery and return the resulting json data.
        /// </summary>
        Task<ITwitterResponse> ExecuteQueryAsync(ITwitterRequest request, ITwitterClientHandler handler = null);

        /// <summary>
        /// Execute a multipart TwitterQuery and return the resulting json data.
        /// </summary>
        Task<ITwitterResponse> ExecuteMultipartQueryAsync(ITwitterRequest request);
    }
}
